docs/usage/REST-API.md

```diff
@@ -14,35 +15,211 @@ Options:
 
-By executing `startleft server` it is possible to see the command-line messages finishing with the following:
+When executing `startleft server` the following command-line message indicates that StartLeft's REST API is ready:
 
 ```Uvicorn running on http://127.0.0.1:5000 (Press CTRL+C to quit)```
 
-so indicating StartLeft's REST API is ready. You can see the endpoints provided by opening the following URL in a 
-web browser: [http://localhost:5000/docs](http://localhost:5000/docs).
+You can see and try the endpoints provided by opening the following URL in a web browser:
+[http://localhost:5000/docs](http://localhost:5000/docs)
 
-Available endpoints:
+### Additional options
+You can also set general StartLeft options when using the API mode. More information on the [Command Line Interface](Command-Line-Interface.md) page 
+or with the command `startleft --help server`.
+
+For example, to launch the server with a log level of DEBUG for testing purposes you can use the following command:
+
+`startleft --log-level DEBUG server`
+
+## Endpoints
+This section describes all the available endpoints, their parameters, and example requests and responses. Refer to each 
+specific format section for example files to use.
+
+### Health
 ```
 GET /health
 ```
+This endpoint can be used to check the status of the application and its [version](../Versioning.md).
+
+Example request:
+```shell
+curl localhost:5000/health
+```
+Example response:
+```json
+{
+    "status": "OK",
+    "version": "1.10.0",
+    "components": {
+        "StartLeft": "OK"
+    }
+}
+```
+### IaC
 ```
 POST /api/v1/startleft/iac
 Request Body:
     iac_file:                   Required. File that contains the IaC definition. If you want to add more than one IaC file, you need to duplicate this tag as many times as files you want to upload.
-    iac_type:                   Required. Type of the IaC File: [CLOUDFORMATION, TERRAFORM]
+    iac_type:                   Required. Type of the IaC file: [CLOUDFORMATION, TERRAFORM]
     id                          Required. ID of the new project
     name                        Required. Name of the new project
     mapping_file                Required. File that contains the mapping between IaC resources and threat model resources.
 ```
+This endpoint accepts one or more IaC source files (currently [Cloudformation](../startleft-processors/iac/cft/CloudFormation-Quickstart.md) 
+or [Terraform](../startleft-processors/iac/tf/Terraform-Quickstart.md)) and a mapping file, and generates an OTM with 
+the resulting threat modeling content.
+
+Example request:
+```shell
+curl --location 
+--request POST localhost:5000/api/v1/startleft/iac \
+--header "Content-Type: multipart/form-data" \
+--header "Accept: application/json" \
+--form iac_type="CLOUDFORMATION" \
+--form iac_file=@"./cft-1.json" \
+--form iac_file=@"./cft-2.json" \
+--form mapping_file=@"./cft-mapping.yaml" \
+--form id="cft-to-otm-example" \
+--form name="CFT to OTM example"
+```
```

Comment:
Maybe we can use real files for the examples, rather than dummy files.

Reply:
Done! I tried to link existing files directly form the repository, but also added the code of some mapping files which are specific to the examples.